tests/Doctrine/Tests/ODM/PHPCR/Functional/Translation/DocumentManagerTest.php

```diff
@@ -881,4 +881,31 @@ public function testAdditionalFindCallsDoNotRefresh()
 
         $this->assertEquals('Guten tag', $trans->topic);
     }
+
+    /**
+     * It should be able to find a bound translation before flush.
+     */
+    public function testRetainTranslation()
+    {
+        $article = new Article;
+        $article->id = '/functional/article-1';
+        $article->text = 'Foo';
+        $this->dm->persist($article);
+
+        $article->topic = 'Hello everybody!';
+        $this->dm->bindTranslation($article, 'en');
+
+        $article->topic = 'Bonjour le monde!';
+        $this->dm->bindTranslation($article, 'fr');
+        $articleInstance = $this->dm->findTranslation(get_class($article), '/functional/article-1', 'en');
```

Comment:
Yeah, I was trying to illustrate the problrem, but yeah, pointlesss.

On 9 October 2015 16:45:53 BST, David Buchmann <notifications@github.com> wrote:
>> +    /**
>> +     * It should be able to find a bound translation before flush.
>> +     */
>> +    public function testRetainTranslation()
>> +    {
>> +        $article = new Article;
>> +        $article->id = '/functional/article-1';
>> +        $article->text = 'Foo';
>> +        $this->dm->persist($article);
>> +
>> +        $article->topic = 'Hello everybody!';
>> +        $this->dm->bindTranslation($article, 'en');
>> +
>> +        $article->topic = 'Bonjour le monde!';
>> +        $this->dm->bindTranslation($article, 'fr');
>> +        $articleInstance =
>$this->dm->findTranslation(get_class($article),
>'/functional/article-1', 'en');
>
>this will be the same object as $article. not sure if we should assign
>in this case. if we have no test about that, maybe assertSame($article,
>$articleInstance).
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/doctrine/phpcr-odm/pull/665/files#r41645115

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.